[부산대 2팀_FE] Release/2025-09-19

byunggil723 opened the pull request:
## 📝 PR 유형
- [ ] 🛠️ Bug Fix (버그 수정)
- [x] ✨ Feature (새로운 기능 추가)
- [ ] ⚙️ Refactor (코드 리팩토링)
- [ ] 📄 Docs (문서 수정)
- [ ] 🪛 Chore (빌드, 설정, 기타 변경)

## 🔧 작업 내용
1. GitHub Actions를 이용해 프론트엔드 정적 파일을 S3 + CloudFront에 자동 배포하는 CI/CD 파이프라인 구축했습니다.
   main 브랜치와 develop 브랜치에 push 혹은 PR Merge 시 자동으로 빌드 및 배포가 실행되도록 설정했습니다.
   우선 main 브랜치에도 develop과 동일한 CI/CD를 설정해 두었으나, 추후에는 실제 운영용 배포 도메인을 별도로 구성할 수도 있을 것 같습니다.
   배포 후 모바일 기기에서 확인해 보니, 모바일 브라우저에서는 주소창 영역에 의해 vh 등의 단위가 웹 브라우저와 다르게 해석되어 최상위 컨테이너의 크기를 모바일 환경에 맞게 조정할 필요가 있었습니다.

> 배포 링크: https://d1xw9usx5fpolc.cloudfront.net

2. 취미 선택 시, 스포츠/관람/보드게임 등 각 카테고리별 15~20개 정도의 취미를 선택할 수 있게 변경했습니다.

<img width="394" height="852" alt="스크린샷 2025-09-19 오후 4 52 55" src="https://github.com/user-attachments/assets/073da934-0099-4117-8b86-ce4d0764377a" />

3. 해시태그를 달 수 있습니다.
   띄어쓰기를 통해 입력하며, 최대 3개까지 7자 제한의 해시태그를 달 수 있습니다.

<img width="394" height="119" alt="스크린샷 2025-09-19 오후 4 54 08" src="https://github.com/user-attachments/assets/c5579f63-18b1-4492-96d7-cec5311747b7" />

4. 모임 시작과 종료 시간을 설정할 수 있습니다.

<img width="495" height="306" alt="스크린샷 2025-09-19 오후 4 59 19" src="https://github.com/user-attachments/assets/e1b6f41e-e954-4e16-bd3d-3a68724308e9" />

5. 모임의 시작 위치를 설정할 수 있습니다.

<img width="394" height="107" alt="스크린샷 2025-09-19 오후 5 00 12" src="https://github.com/user-attachments/assets/d37239a2-cb51-4d7d-94b3-4ad2590d2d2e" />

<br>

<img width="393" height="854" alt="스크린샷 2025-09-19 오후 5 01 49" src="https://github.com/user-attachments/assets/b68801ac-8673-46b3-be1a-26612e3634d4" />

6. 로그인/회원 가입 기능을 구현했습니다.
   카카오 간편 로그인 클릭 시, 카카오 로그인 페이지로 이동합니다.

<img width="394" height="852" alt="스크린샷 2025-09-19 오후 5 03 13" src="https://github.com/user-attachments/assets/920c8d93-9979-45cd-8886-bbe1c6ada7dd" />

<img width="394" height="852" alt="스크린샷 2025-09-19 오후 5 03 34" src="https://github.com/user-attachments/assets/2b34abf1-0f91-44ac-b073-7abfb93a1c5c" />

## 📄 의존성 추가/변경 사항

1. axios 1.12.2 추가

## 💬 기타 사항

중간중간 별도의 PR 없이 생성된 커밋들은, 코드 문제로 인해 중단된 CI/CD를 오류 수정 후 즉시 push한 기록입니다.

## 📂 참고 자료

> N/A

byunggil723 committed:
[FEAT] GitHub Actions 기반 프론트엔드 CI/CD 구축 (#28)

* chore: .gitignore 수정(src/config/apikey.ts 제거) (#26)

* ci: S3 + CloudFront 배포용 GitHub Actions 워크플로우 추가 (#26)

* chore: .nvmrc 추가 및 GitHub Actions CI/CD 파이프라인 Node.js 버전 .nvmrc 값으로 동기화 (#26)

byunggil723 committed:
chore: 불필요한 코드 삭제

byunggil723 committed:
fix: deploy.yml 파일 수정

byunggil723 committed:
fix: deploy.yml 파일 수정(2)

byunggil723 committed:
fix: deploy.yml 파일 수정(3)

bootaeng committed:
[FEAT] 방 생성 페이지 로직 구현 (#27)

* refactor:useEffect 대신 useMemo 사용(#16)

* feat:방 생성 페이지 헤더 컴포넌트 구현(#16)

* feat:공통 스타일 및 기능 컴포넌트화(#16)

* feat:시간 검증 로직 컴포넌트 구현(#16)

* feat:컴포넌트 연결 완료(#16)

* feat:위치 선정 컴포넌트 구현(#16)

* feat:위치설정 버그 수정(#16)

---------

Co-authored-by: bootaeng <tgwon0947@gmail.com>

byunggil723 committed:
fix: 불필요한 코드 삭제 및 export 오류 수정

hyeongchanan committed:
[FEAT] 로그인, 회원가입 API 연결 (#29)

* 카카오 로그인 api (#13)

* 로그인, 회원가입 api 연동 (#13)

* 카카오 로그인 api 수정 (#13)

* 기타 수정 (#13)

byunggil723 committed:
fix: 불필요한 코드 삭제 및 사용되지 않는 변수 임시 사용 조치

june-by commented on src/api/axiosInstance.ts:
```diff
@@ -0,0 +1,37 @@
+  (error) => {
+    // 401 에러 시 토큰 제거
+    if (error.response?.status === 401) {
+      localStorage.removeItem("accessToken");
+      localStorage.removeItem("refreshToken");
```
만들어놓은 clearTokens 유틸함수를 써도 될 것 같습니다!

june-by commented on src/api/axiosInstance.ts:
```diff
@@ -0,0 +1,37 @@
+  },
+  (error) => {
+    // 401 에러 시 토큰 제거
+    if (error.response?.status === 401) {
```
401 이라는 상태값을 상수화하고 네이밍을 통해 의미를 더해주는건 어떨까요?

june-by reviewed (changes requested):
안녕하세요~

한 주 동안 고생하셨습니다!

생각해볼만한 내용이나 수정해야하는 부분들은 코멘트로 남겨놓았습니다~!

bootaeng commented on src/App.tsx:
```diff
@@ -1,11 +1,16 @@
-import MeetingRoom from '@/pages/MeetingRoom';
-import GlobalStyle from '@/style/GlobalStyle';
+
+import Landing from './pages/Landing';
+import Login from './pages/Login';
+import Signup from './pages/Signup';
+import Home from './pages/Home';
+import MeetingRoom from './pages/MeetingRoom';
+import GlobalStyle from './style/GlobalStyle';
+import KakaoLogin from './pages/KakaoLogin';
```
넵 절대경로로 통일했습니다!

bootaeng commented on src/components/room_create_page.tsx/HashtagInput.tsx:
```diff
@@ -0,0 +1,79 @@
+    if (e.key === ' ' && inputValue.trim() !== '') {
+      e.preventDefault();
+      if (hashtags.length < 3) {
+        setHashtags([...hashtags, `#${inputValue.trim()}`]);
```
넵! 수정했습니다!

bootaeng commented on src/components/room_create_page.tsx/HashtagInput.tsx:
```diff
@@ -0,0 +1,79 @@
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    // 7자 글자 수 제한
+    if (e.target.value.length <= 7) {
```
넵! 상수화 시켜서 다루겠습니다!

bootaeng commented on src/components/room_create_page.tsx/HashtagInput.tsx:
```diff
@@ -0,0 +1,79 @@
+        disabled={hashtags.length >= 3}
+      />
+      <HashtagContainer>
+        {hashtags.map((tag) => (
+          <Hashtag key={tag}>
```
넵! 중복되지 않도록 리팩터링 하겠습니다!

bootaeng commented on src/components/room_create_page.tsx/useCreateForm.ts:
```diff
@@ -0,0 +1,98 @@
+  const now = new Date();
+  const start = new Date(startTime);
+  const end = new Date(endTime);
+  const in24Hours = new Date(now.getTime() + 24 * 60 * 60 * 1000);
```
넵! 상수화 하여 재정의 했습니다!

bootaeng committed:
[부산대 2팀_FE] refactor/2025-09-19 (#31)

* refactor:절대경로로 통일

* refactor:setState사용하여 이전값 처리

* refactor:글자수 제한을 상수화

* refactor:prevState활용하여 해시태그 제거

* refactor:해시태그 중복되지 않도록 수정

* refactor:props로 setState 넘기는 부분 수정 + 타입 유틸화

* refactor:in24hours 상수화

* refactor:useEffect 대신 useMemo 사용

* refactor:만들어 놓은 useBoolean 컴포넌트 활용

---------

Co-authored-by: bootaeng <tgwon0947@gmail.com>

hyeongchanan committed:
[부산대 2팀_FE] refactor/2025-09-19 (2) (#32)

* refactor:useLogin 추가 및 기타 수정

1.authUtiles 삭제후 useLogin 으로 대체
2.tokenStorage 함수사용
3.에러코드 명시적으로 표현 및 기타 수정

---------

Co-authored-by: bootaeng <tgwon0947@gmail.com>

june-by commented on src/hooks/useLogin.ts:
```diff
@@ -0,0 +1,83 @@
+  }, []);
+
+  // 이메일 로그인
+  const handleEmailLogin = useCallback(async (email: string, password: string) => {
+    try {
+      const result = await login(email, password);
+      handleLoginSuccess(result, 'email');
+    } catch (error: any) {
+      handleLoginError(error, 'email');
+    }
+  }, [handleLoginSuccess, handleLoginError]);
+
+  // 카카오 로그인
+  const handleKakaoLogin = useCallback(async (code: string) => {
+    try {
+      const result = await kakaoLogin(code);
+      handleLoginSuccess(result, 'kakao');
+    } catch (error: any) {
+      handleLoginError(error, 'kakao');
+    }
+  }, [handleLoginSuccess, handleLoginError]);
+
+  // 회원가입
+  const handleSignup = useCallback(async (email: string, password1: string, password2: string) => {
+    try {
+      const result = await signup(email, password1, password2);
+      handleLoginSuccess(result, 'email');
+    } catch (error: any) {
+      handleLoginError(error, 'email');
+    }
+  }, [handleLoginSuccess, handleLoginError]);
```
useLogin 안에서 너무 많은 것을 담당하는 것 같아요!

handleLoginSuccess, handleLoginError 는 공용 훅으로 분리하고

일반 로그인, 카카오 로그인, 회원가입을 각각의 훅으로 분리해서 공용훅을 불러와서 구현하면

각 훅의 역할이 명확해질것 같아요!

그리고 지금보았는데, 회원가입 로직에서도 handleLoginSuccess, handleLoginError을 사용중인데

회원가입 관련 로직에서 로그인 관련 메서드를 호출하는게 어색하네요!

handleLoginSuccess, handleLoginError 요 네이밍을 회원가입도 포함하는 느낌으로 수정하는것도 괜찮아 보입니다!

june-by reviewed (approved):
로그인 관련 훅에 코멘트 남겨놓았는데, 확인후에 이번주 작업에 반영해주셔도 될 것 같습니다!

고생하셨습니다 👍
